cmd/sfncli/runner.go

```diff
@@ -20,99 +25,170 @@ const maxTaskOutputLength = 32768
 const maxTaskFailureCauseLength = 32768
 
 type TaskRunner struct {
-	sfnapi    sfniface.SFNAPI
-	taskToken string
-	cmd       string
+	sfnapi          sfniface.SFNAPI
+	taskToken       string
+	cmd             string
+	logger          logger.KayveeLogger
+	execCmd         *exec.Cmd
+	receivedSigterm bool
 }
 
 func NewTaskRunner(cmd string, sfnapi sfniface.SFNAPI, taskToken string) TaskRunner {
 	return TaskRunner{
 		sfnapi:    sfnapi,
 		taskToken: taskToken,
 		cmd:       cmd,
+		logger:    logger.New("sfncli"),
 	}
 }
 
-func (t TaskRunner) handleProcessError(
-	ctx context.Context, executionName, title string, err error,
-) error {
-	log.ErrorD(title, logger.M{"error": err.Error(), "execution_name": executionName})
-
-	_, sendErr := t.sfnapi.SendTaskFailureWithContext(ctx, &sfn.SendTaskFailureInput{
-		Cause:     aws.String(err.Error()),
-		TaskToken: &t.taskToken,
-	})
-	if sendErr != nil {
-		return fmt.Errorf("error sending task failure: %s", sendErr)
-	}
-
-	return err
-}
-
-// Process runs the underlying cmd with the appropriate
-// environment and command line params
-func (t TaskRunner) Process(ctx context.Context, args []string, input string) error {
+// Process runs the underlying command.
+// The command inherits the environment of the parent process.
+// Any signals sent to parent process will be forwarded to the command.
+// If the context is canceled, the command is killed.
+func (t *TaskRunner) Process(ctx context.Context, args []string, input string) error {
 	if t.sfnapi == nil { // if New failed :-/
-		return t.handleProcessError(
-			ctx, "process-init", "unknown", fmt.Errorf("NewTaskFailure -- nil sfnapi"),
-		)
+		return t.sendTaskFailure(TaskFailureUnknown{errors.New("nil sfnapi")})
 	}
 
 	var taskInput map[string]interface{}
 	if err := json.Unmarshal([]byte(input), &taskInput); err != nil {
-		return t.handleProcessError(
-			ctx, "process-input", "unknown", fmt.Errorf("Input must be a json object: %s", err),
-		)
+		return t.sendTaskFailure(TaskFailureTaskInputNotJSON{input: input})
+	}
+
+	// convention: if the input contains _EXECUTION_NAME, pass it to the environment of the command
+	var executionName *string
+	if e, ok := taskInput["_EXECUTION_NAME"].(string); ok {
+		executionName = &e
+		t.logger.AddContext("execution_name", *executionName)
```

Comment:
Might want to clear this context out at the end of this function.  It's possible that the next job won't have an execution_name.

Reply:
the logger is specific to a `TaskRunner`, which is created when `sfncli` receives a task, so the logger will only be used for a single task